src/main/java/org/snomed/snowstorm/mrcm/MRCMLoader.java

```diff
@@ -18,116 +21,159 @@
+    }
+
+    private List<AttributeDomain> getAttributeDomains(final String branchPath,
+                                                      final BranchCriteria branchCriteria,
+                                                      final TimerUtil timer) throws ServiceException {
+        List<ReferenceSetMember> attributeDomainMembers = memberService.findMembers(branchPath, branchCriteria,
+                new MemberSearchRequest().referenceSet(Concepts.REFSET_MRCM_ATTRIBUTE_DOMAIN_INTERNATIONAL), LARGE_PAGE).getContent();
+
+        List<AttributeDomain> attributeDomains = new ArrayList<>();
+        for (ReferenceSetMember member : attributeDomainMembers) {
+            // id	effectiveTime	active	moduleId	refsetId	referencedComponentId	domainId
+            // grouped	attributeCardinality	attributeInGroupCardinality	ruleStrengthId	contentTypeId
+            if (member.isActive()) {
+                attributeDomains.add(new AttributeDomain(
+                        member.getMemberId(),
+                        member.getEffectiveTime(),
+                        member.isActive(),
+                        member.getReferencedComponentId(),
+                        member.getAdditionalField("domainId"),
+                        "1".equals(member.getAdditionalField("grouped")),
+                        new Cardinality(member.getAdditionalField("attributeCardinality")),
+                        new Cardinality(member.getAdditionalField("attributeInGroupCardinality")),
+                        RuleStrength.lookupByConceptId(member.getAdditionalField("ruleStrengthId")),
+                        ContentType.lookupByConceptId(member.getAdditionalField("contentTypeId"))
+                ));
+            }
+        }
+        timer.checkpoint("Load attribute domains");
+
+        return attributeDomains;
+    }
+
+    private List<AttributeRange> getAttributeRanges(final String branchPath,
+                                                    final BranchCriteria branchCriteria,
+                                                    final TimerUtil timer) {
+        List<ReferenceSetMember> attributeRangeMembers = memberService.findMembers(branchPath, branchCriteria,
+                new MemberSearchRequest().referenceSet(Concepts.REFSET_MRCM_ATTRIBUTE_RANGE_INTERNATIONAL), LARGE_PAGE).getContent();
+
+        List<AttributeRange> attributeRanges = new ArrayList<>();
+        for (ReferenceSetMember member : attributeRangeMembers) {
+            // id	effectiveTime	active	moduleId	refsetId	referencedComponentId	rangeConstraint	attributeRule	ruleStrengthId	contentTypeId
+            if (member.isActive()) {
+                attributeRanges.add(new AttributeRange(
+                        member.getMemberId(),
+                        member.getEffectiveTime(),
+                        member.isActive(),
+                        member.getReferencedComponentId(),
+                        member.getAdditionalField("rangeConstraint"),
```

Comment:
Could set the concrete type here too, or get the constructor method to set it, if there is one.

Reply:
Good idea - I have added an additional constructor to AttributeRange with a ConcreteValue.DataType field option.